google/cloud/storage/blob.py

```diff
@@ -3053,37 +3059,32 @@ def compose(
             activates it only if certain conditions are met. This class exists to provide safe defaults
             for RPC calls that are not technically safe to retry normally (due to potential data
             duplication or other side-effects) but become safe to retry if a condition such as
-            if_metageneration_match is set.
+            if_generation_match is set.
 
             See the retry.py source code and docstrings in this package (google.cloud.storage.retry) for
             information on retry types and how to configure them.
 
         Example:
-            Compose blobs using generation match preconditions.
+            Compose blobs using source generation match preconditions.
 
             >>> from google.cloud import storage
             >>> client = storage.Client()
             >>> bucket = client.bucket("bucket-name")
 
             >>> blobs = [bucket.blob("blob-name-1"), bucket.blob("blob-name-2")]
-            >>> if_generation_match = [None] * len(blobs)
-            >>> if_generation_match[0] = "123"  # precondition for "blob-name-1"
+            >>> if_source_generation_match = [None] * len(blobs)
+            >>> if_source_generation_match[0] = "123"  # precondition for "blob-name-1"
 
             >>> composed_blob = bucket.blob("composed-name")
-            >>> composed_blob.compose(blobs, if_generation_match)
+            >>> composed_blob.compose(blobs, if_source_generation_match=if_source_generation_match)
         """
         sources_len = len(sources)
```

Comment:
It would seem clearer to map the default value for `if_source_generation_match` as a sequence of `None` values, e.g.:

```python
        if if_source_generation_match is None:
            if_source_generation_match = [None] * sources_len
```
and then do the check:
```python
        if len(if_source_generation_match) != sources_len:
            raise ValueError(...)
```
I would also move both statements down to where they are being used below.

Reply:
Thanks @tseaver! That makes sense. I've made the changes and grouped them closer together. Though wouldn't we want to do the check and raise the exception early in the method? WDYT?

Reply:
@cojenco 
> Wouldn't we want to do the check and raise the exception early in the method?

I like the clarity of doing all the manipulations which depend on `if_source_generation_match` in one place:  in the odd case where the user passes an invalid sequence, we will have done only a trivial amount of work populating `query_params` before raising.